types/webpack/index.d.ts

```diff
@@ -1409,20 +1410,30 @@ declare namespace webpack {
         apply(resolver: any /* EnhancedResolve.Resolver */): void;
     }
 
-    abstract class Stats {
+    class Stats {
         compilation: compilation.Compilation;
         hash?: string;
         startTime?: number;
         endTime?: number;
+
+        static filterWarnings(warnings: string[], warningsFilter?: (warning: string) => boolean): string[];
```

Comment:
I think the type of `warningsFilter` is more complex.
Looking at the code it seems that the filter is always an array which can be a string, RegEx or a custom function: https://github.com/webpack/webpack/blob/v4.42.1/lib/Stats.js#L37
This also aligns with the types from the v5 docs: https://webpack.js.org/configuration/stats/#statswarningsfilter

```suggestion
        static filterWarnings(
          warnings: string[],
          warningsFilter?: Array<string | RegExp | ((warning: string) => boolean)>
        ): string[];
```